lib/logstash/inputs/tcp.rb

```diff
@@ -153,7 +153,7 @@ def handle_socket(socket, client_address, client_port, output_queue, codec)
       codec.decode(read(socket)) do |event|
         event["host"] ||= client_address
         event["port"] ||= client_port
-        event["sslsubject"] ||= socket.peer_cert.subject if @ssl_enable && @ssl_verify
+        event["sslsubject"] ||= socket.peer_cert.subject.to_s if @ssl_enable && @ssl_verify
```

Comment:
It's not really optional and/or debatable, in the end it is required to facilitate serialization and the internal Java event implementation cannot use Ruby objects, which do not/can not support the Java serializable interface. 

And, in any way, it would not be a good idea to divert to that implicit rule, having anything else than string and numeric would not make sense anyway.